actix-router/src/de.rs

```diff
@@ -185,7 +188,11 @@ impl<'de, T: ResourcePath + 'de> Deserializer<'de> for PathDeserializer<'de, T>
                 .as_str(),
             ))
         } else {
-            visitor.visit_str(&self.path[0])
+            let decoded = FULL_QUOTER
+                .with(|q| q.requote(self.path[0].as_bytes()))
+                .unwrap_or_else(|| self.path[0].to_owned());
+
+            visitor.visit_str(&decoded)
```

Comment:
```suggestion
            visitor.visit_string(decoded)
```
to avoid an allocation.